com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/validation/OpenApi3Validator.java

```diff
@@ -42,19 +43,74 @@ private void validateSecuritySchemeReferences(Model model, AbstractNode node, Se
 
             if (node.isObject()) {
                 for (String field : node.asObject().fieldNames()) {
-                    if (securitySchemes.get(field) == null) {
+                    AbstractNode securityScheme = securitySchemes.get(field);
+
+                    if (securityScheme == null) {
                         errors.add(
                                 error(node.get(field), IMarker.SEVERITY_ERROR, Messages.error_invalid_reference_type));
+                    } else {
```

Comment:
I think here (on line 50), we have enough info about the expected type, so we can add an explanation at the end of the standard "Invalid object reference, the referenced object is not of expected type":
> Invalid object reference, the referenced object is not of expected type. It should be a valid security scheme.